packages/ts/generator-typescript-plugin-model/src/ModelSchemaProcessor.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  isNullableSchema,
+  Schema,
+  decomposeSchema,
+  isComposedSchema,
+  NonComposedSchema,
+  isReferenceSchema,
+  isNumberSchema,
+  isIntegerSchema,
+  isStringSchema,
+  isMapSchema,
+  isArraySchema,
+  isBooleanSchema,
+  convertReferenceSchemaToPath,
+  convertReferenceSchemaToSpecifier,
+  ReferenceSchema,
+  ArraySchema,
+  MapSchema,
+  NonComposedRegularSchema,
+  isNonComposedRegularSchema,
+} from '@vaadin/generator-typescript-core/Schema.js';
+import PluginError from '@vaadin/generator-typescript-utils/PluginError.js';
+import type DependencyManager from '@vaadin/generator-typescript-utils/dependencies/DependencyManager';
+import type { ArrayLiteralExpression, Expression, Identifier, TypeNode } from 'typescript';
+import ts from 'typescript';
+
+type AnnotatedSchema = NonComposedRegularSchema & Readonly<{ 'x-annotations': ReadonlyArray<string> }>;
+
+function isAnnotatedSchema(schema: Schema): schema is AnnotatedSchema {
+  return isNonComposedRegularSchema(schema) && 'x-annotations' in schema;
+}
+
+type AnnotationPrimitiveArgument = boolean | number | string;
+type AnnotationArgument = AnnotationPrimitiveArgument | AnnotationNamedArguments;
+interface AnnotationNamedArguments {
+  readonly [name: string]: AnnotationArgument;
+}
+
+interface Annotation {
+  simpleName: string;
+  arguments: ReadonlyArray<AnnotationArgument>;
+}
+
+function parseAnnotation(annotationText: string): Annotation {
+  const [, simpleName, argumentsText] = /^(\w+)\((.*)\)$/.exec(annotationText) || [];
```

Comment:
Wasn't it supposed to be changed to a JSON format instead of text parsing?

Reply:
Will address in https://github.com/vaadin/fusion/pull/171